### DISCUSSION TITLE
How to integrate with unity?

### DISCUSSION THREAD

**Discussion by Stiven53** [ptitSeb/box64#391] [Q&A]
I have x64 library in project and i want load this library on arm64. How do that? c# sample import dll in script:
```
	[DllImport("UnityUWE")]
	public static extern void FloodSimStep(ref byte flowGrid, ref float valueGrid, Int3 arrayDims, Int3 usedDims, float dt);
```

**Comment by ptitSeb** [comment]
That would require box64 to build as a library, and acting as a proxy for the x86 version of the lib.
I have done some experiment with box86 on this concept (and it worked), but I haven't done anything about this on box64. It's possible with some work, but not done yet.

**Reply by Stiven53** [reply]
and since you have done this, you can send a sample code for the proxy?

**Reply by Stiven53** [reply]
so can you send?

**Reply by ptitSeb** [reply]
not on box64 for now. But take a look at box86, and build as lib to have a idea.
